pkg/fqdn/cache.go

```diff
@@ -170,15 +168,18 @@ func NewDNSCacheWithLimit(limit int) *DNSCache {
 // ips may be an IPv4 or IPv6 IP. Duplicates will be removed.
 // ttl is the DNS TTL for ips and is a seconds value.
 func (c *DNSCache) Update(lookupTime time.Time, name string, ips []net.IP, ttl int) bool {
+	c.Lock()
+	if c.minTTL > 0 && ttl < c.minTTL {
```

Comment:
Since minTTL is a constant, do we need to take the lock? In any case, if you move the lock here could you also move the defer unlock?

Reply:
@raybejjani fixed.